cmd/terramate/cli/cloud_credential_github_oidc.go

```diff
@@ -177,11 +182,22 @@ func (g *githubOIDC) Token() (string, error) {
 	return g.token, nil
 }
 
+func (g *githubOIDC) ApplyCredentials(req *http.Request) error {
```

Comment:
You can have a single implementation composable for OIDC credentials that implement these new methods to avoid duplicating code. I'm thinking we could potentially add a new OIDC client in near future.

Reply:
makes sense. Thanks for the idea!

Reply:
You could remove this by using a type composition:

type jwtCredentials struct {}
func (c jwtCredentials) ApplyCredentials(req *http.Request) error {
...
type githubOIDC struct {
      jwtCredentials
      ...
}

But it is fine too this way without the power of Go :D

Reply:
that was my idea when you proposed it but the implementation needs access to the credential `Token()` method.
See here: https://github.com/terramate-io/terramate/pull/2001/files#diff-c3ea63583f653692282c68c1bf245583b66f17ed6a0863ba92fc42f0cc69aa98R22

so it requires a bit of setup to have the jwtCredentials instantiated with the credential or change the interface to be `ApplyCredentials(req, cred)`.
The functions seemed to be less code and less error prone.